services/workflows-service/src/auth/local/local-auth.guard.ts

```diff
@@ -1,35 +1,27 @@
+import { Injectable } from '@nestjs/common';
 import { AuthGuard } from '@nestjs/passport';
 import { ExecutionContext } from '@nestjs/common';
-import type { Request } from 'express';
-import { createClient } from '@supabase/supabase-js';
-import { env } from '@/env';
+import { SupabaseService } from '../../supabase/supabase.service';
 
+@Injectable()
 export class LocalAuthGuard extends AuthGuard('local') {
-  async canActivate(context: ExecutionContext) {
-    const result = await super.canActivate(context);
-    const request = context.switchToHttp().getRequest<Request>();
-    await super.logIn(request);
-    if (env.TELEMETRY_ENABLED && env.TELEMETRY_SUPABASE_URL && env.TELEMETRY_SUPABASE_API_KEY) {
-      try {
-        const SupabaseClient = createClient(
-          env.TELEMETRY_SUPABASE_URL,
-          env.TELEMETRY_SUPABASE_API_KEY,
-          {
-            db: { schema: 'public' },
-          },
-        );
-        const fullUrl = `${request.protocol}://${request.get('Host')}${request.originalUrl}`;
-        const { data: result, error } = await SupabaseClient.from('logins').insert([
-          { url: fullUrl },
-        ]);
-        if (error) {
-          console.error('Error inserting data:', error.message);
-          return false;
-        }
-      } catch (err) {
-        console.error('Unexpected error:', err);
-      }
+  constructor(private readonly supabaseService: SupabaseService) {
+    super();
+  }
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const result = (await super.canActivate(context)) as boolean;
+    const request = context.switchToHttp().getRequest();
+
+    // Use the injected service
+    if (result && request.user) {
+      const fullUrl = request.protocol + '://' + request.get('host') + request.originalUrl;
+      await this.supabaseService.logSignIn(fullUrl);
```

Comment:
@MatanYadaev @alonp99 Whenever you do `await` it will wait till the login in operation return with response and it wont continue since we've got response from supabase client.

i think we shouldnt wait in here what do you think?

Reply:
youre right